syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/datagenerator/RecordCreator.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.syndeo.transforms.datagenerator;
+
+import com.google.common.flogger.GoogleLogger;
+import java.util.List;
+import java.util.Random;
+import org.apache.avro.Schema;
+import org.apache.avro.Schema.Field;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.GenericRecordBuilder;
+import org.apache.beam.sdk.schemas.utils.AvroUtils;
+import org.apache.beam.sdk.values.Row;
+import org.apache.beam.sdk.values.Row.FieldValueBuilder;
+
+public final class RecordCreator {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static final String ALPHA_NUMBERIC_STRING =
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789abcdefghijklmnopqrstuvxyz";
+
+  public static Random random = new Random();
+
+  public static Row createRowRecord(Schema schema) {
+    org.apache.beam.sdk.schemas.Schema beamSchema = AvroUtils.toBeamSchema(schema);
+    Row.Builder builder = Row.withSchema(beamSchema);
+    FieldValueBuilder fvbuilder = null;
+    for (Field field : schema.getFields()) {
+      if (field.schema().getType() != Schema.Type.RECORD) {
+        Object value = generateRandomValue(field);
+        if (fvbuilder == null) {
+          fvbuilder = builder.withFieldValue(field.name(), value);
+        } else {
+          fvbuilder = fvbuilder.withFieldValue(field.name(), value);
+        }
+      } else {
+        if (fvbuilder == null) {
+          fvbuilder = builder.withFieldValue(field.name(), createGenericRecord(field.schema()));
```

Comment:
this is a little confusing to me. We are appending an Avro `GenericRecord` to a Beam `Row`. This is not supported. Have you tested that a nested record will work?